test/new-e2e/tests/containers/kindvm_test.go

```diff
@@ -8,27 +8,37 @@ package containers
 import (
 	"context"
 	"fmt"
-	"os"
-	"path"
-	"testing"
-
 	fakeintake "github.com/DataDog/datadog-agent/test/fakeintake/client"
 	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/runner"
 	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/infra"
 	"github.com/DataDog/test-infra-definitions/scenarios/aws/kindvm"
-
 	"github.com/pulumi/pulumi/sdk/v3/go/auto"
 	"github.com/stretchr/testify/suite"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/tools/clientcmd"
+	"os"
+	"path"
+	"strings"
+	"testing"
 )
 
 type kindSuite struct {
 	k8sSuite
+	k8sVersion string
+	stackName  string
 }
 
 func TestKindSuite(t *testing.T) {
-	suite.Run(t, &kindSuite{})
+
+	KubeVersions := []string{"1.27", "1.23", "1.20", "1.19"}
+
+	for _, version := range KubeVersions {
+		version := version
```

Comment:
Is this intended? I don't see its purpose
```
	KubeVersions := []string{"1.27", "1.23", "1.20", "1.19"}

	for _, version := range KubeVersions {
		fmt.Printf("%s, %T\n", version, version)
		version := version
		fmt.Printf("%s, %T\n", version, version)
	}
```
Returns
```
1.27, string
1.27, string
1.23, string
1.23, string
1.20, string
1.20, string
1.19, string
1.19, string
```

Reply:
The purpose is capturing the range iterator because we are launching a goroutine within the for loop. 

If we don't do so, the range variable might be modified and multiple go routines might get the same value for the `version`

Reply:
Check [this](https://eli.thegreenplace.net/2019/go-internals-capturing-loop-variables-in-closures/) for more details

Reply:
Got it, thanks for the explanation 👍